docs/accessing-services.asciidoc

```diff
@@ -0,0 +1,157 @@
+
+A user with the `superuser` role is managed by the operator to access to Elasticsearch and Kibana. Its password is stored in a `Secret` named `<name>-elastic-user`.
+
+[source,sh]
+----
+> kubectl get secret hulk-elastic-user -o go-template='{{.data.elastic | base64decode }}'
+5abc6x1234bb7vbabc4xyz42
+----
+
+[id="{p}-allow-public-access"]
+== Allow public access
+
+Services can be exposed in link:https://kubernetes.io/docs/concepts/services-networking/service/#publishing-services-service-types[different ways] by specifying an `http.service.spec.type` in the `spec` of a resource manifest.
+On cloud providers which support external load balancers, setting the `type` field to `LoadBalancer` will provision a load balancer for the `Service` and populate the column `EXTERNAL-IP` after a short delay (depending the cloud provider, it may incur charges):
+
+[source,yaml]
+----
+apiVersion: <kind>.k8s.elastic.co/v1alpha1
+kind: <Kind>
+metadata:
+  name: hulk
+spec:
+  version: 7.1.0
+  http:
+    service:
+      spec:
+        type: LoadBalancer
+----
+
+[source,sh]
+----
+> kubectl get svc
+
+NAME                TYPE           CLUSTER-IP      EXTERNAL-IP      PORT(S)          AGE
+hulk-es             LoadBalancer   10.19.252.160   35.198.131.115   9200:32361/TCP   25m
+hulk-kibana         LoadBalancer   10.19.247.151   35.242.197.228   5601:32567/TCP   25m
+----
+
+[id="{p}-static-ip-custom-domain"]
+== Static IP and custom domain
+
+To use a custom domain name, you may have to reserve a static IP and/or use an Ingress instead of a `LoadBalancer` `Service` depending your needs and what your cloud provider provides. Whatever is used, your DNS must be added to the certificate SAN in the `spec` of your Elastic resources manifest:
+
+
+[source,yaml]
+----
+spec:
+  http:
+    service:
+      spec:
+        type: LoadBalancer
+    tls:
+      selfSignedCertificate:
+        subjectAltNames:
+        - dns: hulk.example.com
+----
+
+[id="{p}-request-elasticsearch-endpoint"]
+== Request the Elasticsearch endpoint
+
+To request the Elasticsearch endpoint using basic authentication and certificate validation with the CA, you
+have to retrieve the CA certificate, the IP of the service and the password of the elastic user:
+
+[source,sh]
+----
+NAME=hulk
+
+CA=$(kubectl get secret "$NAME-ca" -o go-template='{{index .data "ca.pem" | base64decode }}') > ca.pem
+IP=$(kubectl get svc "$NAME-es" -o jsonpath='{.status.loadBalancer.ingress[].ip}')
+PW=$(kubectl get secret "$NAME-elastic-user" -o go-template='{{.data.elastic | base64decode }}')
+
+curl --cacert ca.pem -u elastic:$PW https://$IP:9200/
+----
+
+At this stage, you should encounter this issue:
+
+[source,sh]
+----
+curl: (51) SSL: no alternative certificate subject name matches target host name '35.198.131.115'
+
+----
+
+This is corrected by adding the external IP of the service to the SANs of the certificate (link:https://github.com/elastic/cloud-on-k8s/issues/910[until this is done automatically]):
```

Comment:
Should we wait until we have that?

Reply:
Let's maybe remove the link for now, and fix the doc once we have that available?

Reply:
I feel the need to document that even if it's not glorious. We already have questions around that. Yes, let's remove the link.